CONTRIBUTING.md

```diff
@@ -96,6 +96,64 @@ Closes #123
 
 In this case 123 is the corresponding issue number.
 
+### When End-To-End Tests Fail
+
+Every pull request triggers a full run of testing in the CI system.
+The failures reported by the code style checker (aka the "linter") and the unit tests are usually
+clear and easy to fix (and can be avoided by running `npm t` locally before creating a commit).
+But when an integration, or e2e test, fails, it's sometimes useful to consult the log files
+for the run.
+
+Here's how to see more details for a failed run:
+
+1. The test results can be found by first going to the GitHub page for the PR (for an example, find an open PR at https://github.com/rancher-sandbox/rancher-desktop/pulls?q=is%3Aopen+is%3Apr),
+then look for the list of checks, find the `E2E/Integration Tests` entry, and click on the `Details` link for the first view.
+
+![Failing E2E tests link](./docs/assets/images/cilogs/screenshot01.png)
+
+This takes you to a more detailed view of the test, and which tests failed, but the log files are a few more clicks away.
+
+2. Scroll to the bottom and click on the link that reads `View more details on Cirrus CI`.
+
+![Link to Cirrus CI](./docs/assets/images/cilogs/screenshot02.png)
+3. Once the full test run has completed, the `Artifacts` button acts like a file-system browser.
```

Comment:
Please add a line break after the picture — it looks like GitHub is just doing this inline, which is pretty bad formatting.  You may need to indent the picture to get proper ordered list parsing.

Reply:
Hope the line breaks are enough. The list-ordering is still working without indenting the images.

Reply:
The important thing is the list, not the indent, so it's fine.